packages/dd-trace/src/appsec/waf/waf_manager.js

```diff
@@ -51,20 +54,27 @@ class WAFManager {
     return wafContext
   }
 
-  update (newRules) {
-    try {
-      this.ddwaf.update(newRules)
+  setRulesVersion () {
+    if (this.ddwaf.diagnostics.ruleset_version) {
+      this.rulesVersion = this.ddwaf.diagnostics.ruleset_version
+    }
+  }
 
-      if (this.ddwaf.diagnostics.ruleset_version) {
-        this.rulesVersion = this.ddwaf.diagnostics.ruleset_version
-      }
+  setAsmDdFallbackConfig () {
+    if (!this.ddwaf.configPaths.some(cp => cp.includes('ASM_DD'))) {
+      this.updateConfig(WAFManager.defaultWafConfigPath, this.defaultRules)
+    }
+  }
 
-      Reporter.reportWafUpdate(this.ddwafVersion, this.rulesVersion, true)
-    } catch (error) {
-      Reporter.reportWafUpdate(this.ddwafVersion, 'unknown', false)
+  updateConfig (path, rules) {
+    const updateResult = this.ddwaf.createOrUpdateConfig(rules, path)
+    this.setRulesVersion()
+    return updateResult
+  }
```

Comment:
you know instead of returning updateResult and doing a weird WafUpdateError system in the layer above, you could just return null if success, and the diagnostics if not success ? hence you can remove all the error type checking in the catch ?

Reply:
Yes, this was one of the approaches I played with. However, throwing a WafUpdateError on failed update result allows me to centralize all the error handling for RC (apply_error) in one place.